addons/html_editor/static/src/main/chatgpt/chatgpt_plugin.js

```diff
@@ -123,21 +171,60 @@ export class ChatGPTPlugin extends Plugin {
                     divContainer.prepend(div);
                     setTimeout(() => div.remove(), 2000);
                 }
+                unwrapContents(insertedNodes[0]);
             },
             ...params,
         };
         dialogParams.baseContainer = this.dependencies.baseContainer.getDefaultNodeName();
         // collapse to end
         const sanitize = this.dependencies.sanitize.sanitize;
-        if (selection.isCollapsed) {
-            this.dependencies.dialog.addDialog(ChatGPTPromptDialog, { ...dialogParams, sanitize });
-        } else {
+        if (params.language) {
             const originalText = selection.textContent() || "";
             this.dependencies.dialog.addDialog(
-                params.language ? ChatGPTTranslateDialog : ChatGPTAlternativesDialog,
+                ChatGPTTranslateDialog,
                 { ...dialogParams, originalText, sanitize }
             );
-        }
+        } else {
+            let callerComp, recordModel, recordId, recordData, recordFields, callerId, placeholderPrompt, textSelection;
+            const { resModel, resId, data, fields, id } = this.config.getRecordInfo();
+            if (selection.isCollapsed) {
+                if (resModel === "mail.compose.message") {
+                    callerComp = "html_field_composer";
+                    recordModel = data.model;
+                    recordId = Number(data.res_ids.slice(1,-1));  // resIds should look like so `[id]`, the slice and cast allows to extact the id
+                    recordData = data;
+                    callerId = id;
+                    placeholderPrompt =  _t("Write a followup answer");
+                } else {
+                    callerComp = "html_field_record";
+                    recordModel = resModel;
+                    recordId = resId;
+                    recordData = data;
+                    recordFields = fields;
+                    callerId = resId || id;
+                }
+            } else {
+                callerComp = "html_field_text_select";
+                recordData = data;
+                recordFields = fields;
+                callerId = resId || id;
+                placeholderPrompt = _t("Rewrite");
+                textSelection = selection.textContent();
+            }
+            await this.services.aiChatLauncher.openAIChatFromContextV2({
```

Comment:
is there a dependency from html_editor to ai_apps? is that something we want?

Reply:
Maybe the ai editor plugins should be moved to the ai module.